Reducing args to 5 will require a huge amount of work. But it's a moot point. This isn't special to arg6.

What I am seeing is that passing in pointers to memory for any argument, the compiler gets confused. I think the u32_t cast may have something to with it. For example, I was trying to add support for 64-bit return values, for APIs like k_uptime_get():

```
static inline u64_t _syscall_ret64_invoke0(u32_t call_id)
{
	u64_t ret;

	_arch_syscall_invoke1((u32_t)&ret, call_id);
	return ret;
}
```

I get the following warning, even though I'm passing the address of ret in as a pointer (which the handler verifies and fills in with a 64-bit value)

```
/projects/zephyr/tests/kernel/threads/lifecycle/thread_init/outdir/qemu_x86/include/generated/syscalls/kernel.h: In function ‘k_uptime_get’:
/projects/zephyr/include/syscall.h:251:9: warning: ‘ret’ may be used uninitialized in this function [-Wmaybe-uninitialized]
  return ret;
         ^~~
/projects/zephyr/include/syscall.h:248:8: note: ‘ret’ was declared here
  u64_t ret;
```

The warning goes away if I make the ret parameter volatile. I think this is the same issue and it is unrelated to the use of EBP.

--- a/include/arch/x86/arch.h
+++ b/include/arch/x86/arch.h
@@ -576,7 +576,8 @@ static inline u32_t _arch_syscall_invoke6(u32_t arg1, u32_t arg2, u32_t arg3,
 #endif
 			 : "S" (call_id), "a" (arg1), "d" (arg2),
 			   "c" (arg3), "b" (arg4), "D" (arg5),
-			   [arg6] "m" (arg6));
+			   [arg6] "m" (arg6)
+			 : "memory");
 	return ret;
 }
 
@@ -591,7 +592,8 @@ static inline u32_t _arch_syscall_invoke5(u32_t arg1, u32_t arg2, u32_t arg3,
 			   , "=d" (arg2), "=c" (arg3)
 #endif
 			 : "S" (call_id), "a" (arg1), "d" (arg2),
-			   "c" (arg3), "b" (arg4), "D" (arg5));
+			   "c" (arg3), "b" (arg4), "D" (arg5)
+			 : "memory");
 	return ret;
 }
 
@@ -606,7 +608,8 @@ static inline u32_t _arch_syscall_invoke4(u32_t arg1, u32_t arg2, u32_t arg3,
 			   , "=d" (arg2), "=c" (arg3)
 #endif
 			 : "S" (call_id), "a" (arg1), "d" (arg2), "c" (arg3),
-			   "b" (arg4));
+			   "b" (arg4)
+			 : "memory");
 	return ret;
 }
 
@@ -620,7 +623,8 @@ static inline u32_t _arch_syscall_invoke3(u32_t arg1, u32_t arg2, u32_t arg3,
 #ifdef CONFIG_X86_IAMCU
 			   , "=d" (arg2), "=c" (arg3)
 #endif
-			 : "S" (call_id), "a" (arg1), "d" (arg2), "c" (arg3));
+			 : "S" (call_id), "a" (arg1), "d" (arg2), "c" (arg3)
+			 : "memory");
 	return ret;
 }
 
@@ -634,8 +638,9 @@ static inline u32_t _arch_syscall_invoke2(u32_t arg1, u32_t arg2, u32_t call_id)
 			   , "=d" (arg2)
 #endif
 			 : "S" (call_id), "a" (arg1), "d" (arg2)
+			 : "memory"
 #ifdef CONFIG_X86_IAMCU
-			 : "ecx"
+			 , "ecx"
 #endif
 			 );
 	return ret;
@@ -648,8 +653,9 @@ static inline u32_t _arch_syscall_invoke1(u32_t arg1, u32_t call_id)
 	__asm__ volatile("int $0x80"
 			 : "=a" (ret)
 			 : "S" (call_id), "a" (arg1)
+			 : "memory"
 #ifdef CONFIG_X86_IAMCU
-			 : "edx", "ecx"
+			 , "edx", "ecx"
 #endif
 			 );
 	return ret;
@@ -662,8 +668,9 @@ static inline u32_t _arch_syscall_invoke0(u32_t call_id)
 	__asm__ volatile("int $0x80"
 			 : "=a" (ret)
 			 : "S" (call_id)
+			 : "memory"
 #ifdef CONFIG_X86_IAMCU
-			 : "edx", "ecx"
+			 , "edx", "ecx"
 #endif
 			 );
 	return ret;

